genesyscloud/resource_genesyscloud_routing_queue.go

```diff
@@ -595,22 +595,25 @@ func readQueue(ctx context.Context, d *schema.ResourceData, meta interface{}) di
 		d.Set("media_settings_chat", nil)
 		d.Set("media_settings_email", nil)
 		d.Set("media_settings_message", nil)
-		if currentQueue.MediaSettings != nil {
-			if callSettings, ok := (*currentQueue.MediaSettings)[mediaSettingsKeyCall]; ok {
-				d.Set("media_settings_call", flattenMediaSetting(callSettings))
-			}
-			if callbackSettings, ok := (*currentQueue.MediaSettings)[mediaSettingsKeyCallback]; ok {
-				d.Set("media_settings_callback", flattenMediaSetting(callbackSettings))
-			}
-			if chatSettings, ok := (*currentQueue.MediaSettings)[mediaSettingsKeyChat]; ok {
-				d.Set("media_settings_chat", flattenMediaSetting(chatSettings))
-			}
-			if emailSettings, ok := (*currentQueue.MediaSettings)[mediaSettingsKeyEmail]; ok {
-				d.Set("media_settings_email", flattenMediaSetting(emailSettings))
-			}
-			if messageSettings, ok := (*currentQueue.MediaSettings)[mediaSettingsKeyMessage]; ok {
-				d.Set("media_settings_message", flattenMediaSetting(messageSettings))
-			}
+
+		if currentQueue.MediaSettings != nil && currentQueue.MediaSettings.Call != nil {
```

Comment:
This was a bit of screwery.  The JSON body did not change, but the API owners moved the swagger def from being a map to a specific object with values for each object.  So I had to go through and change some of the code for the routing queues to move away from using a map to using individual objects.